app/pages/sign-up.tsx

```diff
@@ -17,38 +54,94 @@ const SignUp = () => {
     setBenefitRefs((prev) => [...prev, { current: null }]);
   }, []);
 
-  const handleCreateAccount = useCallback(() => {
+  // Only call if wallet connected.
+  const handleCreateAccount = useCallback(async () => {
     const username = usernameRef?.current?.value;
     const email = emailRef?.current?.value;
     const description = descriptionRef?.current?.value;
 
-    const benefits = benefitRefs.map((benefitRef) => {
+    let benefits = benefitRefs.map((benefitRef) => {
       return benefitRef?.current?.value;
     });
 
+    // Filter out blank benefits
+    benefits = benefits.filter((benefit) => benefit);
     const numBenefits = benefits.length;
 
-    console.log('username', username);
-    console.log('email', email);
-    console.log('description', description);
-    console.log('benefits', benefits);
-    console.log('numBenefits', numBenefits);
+    // Create account on chain
+    const creatorSeeds = [
+      connectedWallet!.publicKey.toBuffer(),
+      anchor.utils.bytes.utf8.encode('creator'),
+    ];
+
+    const [creatorPubKey] = await anchor.web3.PublicKey.findProgramAddress(
+      creatorSeeds,
+      program!.programId
+    );
+
+    // Create Creator account
+    const txn = new anchor.web3.Transaction();
+    txn.add(
+      program!.instruction.createAccount(
+        username!,
+        email!,
+        description!,
```

Comment:
there may be a better way to handling the typing here. If these are nullable values, maybe we should handle the case when they are null by showing the user "This field is invalid" or something. 

Not important for this PR, but something we may want to get back to

Reply:
You are completely right. Just wanted to get the progress in to unblock people. I just made an issue